modules/apps/portal/portal-http-impl/src/main/java/com/liferay/portal/http/internal/HttpImpl.java

```diff
@@ -374,6 +375,21 @@ else if (_PROXY_AUTH_TYPE.equals("ntlm")) {
 			_proxyCredentials = null;
 		}
 
+		PoolingHttpClientConnectionManager poolingHttpClientConnectionManager =
```

Comment:
@ericyanLr Please move the setting logic to poolingHttpClientConnectionManager to _createPoolingHttpClientConnectionManager

Reply:
Hi @tinatian ,

This one is a bit tricky, because of the singleton: **_poolingHttpClientConnectionManagerDCLSingleton**. This singleton is not cleared when the OSGI configuration is modified.

- And so, moving this logic to **_createPoolingHttpClientConnectionManager** would result in the new OSGI configuration (tcpKeepAliveEnabled) not being properly used if the singleton instance already exists. 
- And if we attempt to destroy the **_poolingHttpClientConnectionManagerDCLSingleton**, this would result in existing connections being destroyed, and so we did not go this route.